src/registrar/admin.py

```diff
@@ -1856,11 +1856,15 @@ def save_model(self, request, obj, form, change):
         if not change:
             domain = obj.domain
             domain_org = getattr(domain.domain_info, "portfolio", None)
-            requested_email = obj.email
+            requested_email = obj.email.lower()
             # Look up a user with that email
             requested_user = get_requested_user(requested_email)
             requestor = request.user
 
+            # set object email to appropiate user email if it exists
+            if requested_user and requested_user.email:
+                obj.email = requested_user.email
```

Comment:
I believe you want this to be `requested_email = requested_user.email` since below in the `DomainInvitation` create it uses `requested_email`